src/matrix/kaldi-matrix.cc

```diff
@@ -403,19 +403,23 @@ void MatrixBase<Real>::AddSmat(Real alpha, const SparseMatrix<Real> &A,
   if (trans == kNoTrans) {
     KALDI_ASSERT(NumRows() == A.NumRows());
     KALDI_ASSERT(NumCols() == A.NumCols());
-    for (int i = 0; i < A.NumRows(); ++i) {
+    MatrixIndexT a_num_rows = A.NumRows();
+    for (MatrixIndexT i = 0; i < a_num_rows; ++i) {
       const auto & row = A.Row(i);
-      for (int id = 0; id < row.NumElements(); ++id) {
+      MatrixIndexT num_elems = row.NumElements();
+      for (MatrixIndexT id = 0; id < num_elems; ++id) {
         (*this)(i, row.GetElement(id).first) += alpha
             * row.GetElement(id).second;
       }
     }
   } else {
     KALDI_ASSERT(NumRows() == A.NumCols());
     KALDI_ASSERT(NumCols() == A.NumRows());
-    for (int i = 0; i < A.NumRows(); ++i) {
+    MatrixIndexT a_num_rows = A.NumRows();
+    for (MatrixIndexT i = 0; i < a_num_rows; ++i) {
       const auto & row = A.Row(i);
```

Comment:
this is a pre-existing issue in the old code, but we prefer not to use 'auto' in cases like this; use the actual type.  And no space between & and row.